container/src/container_state.rs

```diff
@@ -0,0 +1,187 @@
+use crate::{Error, Result};
+use serde::{Deserialize, Serialize};
+use std::fs;
+use std::fs::{File, OpenOptions};
+use std::io::Write;
+use std::path::PathBuf;
+
+const KAPS_ROOT_PATH: &str = "/var/run/kaps";
+const OCI_VERSION: &str = "0.2.0";
+const STATE_FILE: &str = "state.json";
+
+/// Container runtime status
+#[derive(Serialize, Deserialize, Debug, PartialEq)]
+pub enum Status {
+    #[serde(rename = "creating")]
+    Creating,
+    #[serde(rename = "created")]
+    Created,
+    #[serde(rename = "running")]
+    Running,
+    #[serde(rename = "stopped")]
+    Stopped,
+}
+
+impl Default for Status {
+    fn default() -> Self {
+        Status::Creating
+    }
+}
+
+/// Represent the state of the running container.
+#[derive(Serialize, Deserialize)]
+pub struct ContainerState {
+    id: String,
+    /// Path where the state of the container will be saved.
+    #[serde(skip)]
+    container_path: PathBuf,
+    /// OCI version.
+    oci_version: String,
+    /// Runtime state of the container.
+    pub status: Status,
+    /// ID of the container process.
+    pub pid: i32,
+    /// Path to the bundle.
+    bundle: PathBuf,
+}
+
+impl Default for ContainerState {
```

Comment:
Is this implementation used anywhere?

Reply:
I use de Default trait implementation in the new methods with the  `..Default::default()` to specify missing attributes.

Maybe it's useless in the case that I use it only in this method?

Reply:
> Maybe it's useless in the case that I use it only in this method?

No, that's fine then.